src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java

```diff
@@ -505,11 +511,15 @@ public Mono<ShipmentTO> getShipmentByCarrierBookingReference(
                           .doOnNext(shipmentTO::setShipmentCutOffTimes),
                       fetchShipmentLocationsByBookingID(shipment.getBookingID())
                           .doOnNext(shipmentTO::setShipmentLocations),
-                      fetchCarrierClausesByShipmentID(shipment.getShipmentID())
+                      carrierClauseService
+                          .fetchCarrierClausesByShipmentID(shipment.getShipmentID())
```

Comment:
Local function has a null check, but the service implementation doesn't. Is this going to be a problem?

Reply:
I don't believe so, shipmentID is the primary key column of the shipmentTable so the db constraints will ensure no shipments exist without an ID